java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/dao/JobIdGenerator.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 4Paradigm
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com._4paradigm.openmldb.taskmanager.dao;
+
+import com._4paradigm.openmldb.common.zk.ZKClient;
+import com._4paradigm.openmldb.common.zk.ZKConfig;
+import com._4paradigm.openmldb.taskmanager.config.TaskManagerConfig;
+
+import java.nio.charset.StandardCharsets;
+
+public class JobIdGenerator {
+    private static int maxJobId = 0;
+    private static int jobId = 0;
+    private volatile static ZKClient zkClient;
+
+    private JobIdGenerator() {
+    }
+
+    public static int getUniqueJobID() throws Exception {
+        synchronized (JobIdGenerator.class) {
+            // TODO: Remove these codes, if we support to initialize zkClient when start
+            if (zkClient == null) {
+                zkClient = new ZKClient(ZKConfig.builder()
+                        .cluster(TaskManagerConfig.ZK_CLUSTER)
+                        .namespace(TaskManagerConfig.ZK_ROOT_PATH)
+                        .sessionTimeout(TaskManagerConfig.ZK_SESSION_TIMEOUT)
+                        .baseSleepTime(TaskManagerConfig.ZK_BASE_SLEEP_TIME)
+                        .connectionTimeout(TaskManagerConfig.ZK_CONNECTION_TIMEOUT)
+                        .maxConnectWaitTime(TaskManagerConfig.ZK_MAX_CONNECT_WAIT_TIME)
+                        .maxRetries(TaskManagerConfig.ZK_MAX_RETRIES)
+                        .build());
+                if (!zkClient.connect()) {
+                    throw new RuntimeException("Fail to connect zookeeper!");
+                }
+                // If node exists, will not to create node.
+                zkClient.createNode(
+                        TaskManagerConfig.ZK_ROOT_PATH, "".getBytes());
+                zkClient.createNode(
+                        TaskManagerConfig.ZK_TASKMANAGER_PATH, "".getBytes());
+                zkClient.createNode(
+                        TaskManagerConfig.ZK_MAX_JOB_ID_PATH, String.valueOf(TaskManagerConfig.PREFETCH_JOB_NUM).getBytes());
+                // Get the node value from zk
+                maxJobId = Integer.parseInt(zkClient.getNodeValue(TaskManagerConfig.ZK_MAX_JOB_ID_PATH));
+                jobId = maxJobId;
+            }
+            // Pre-set job id in zookeeper.
+            // If taskmanager failed, pre-set job id in zk can avoid duplicate ID.
+            if ((jobId + 1) >= maxJobId) {
+                maxJobId = jobId + 1 + maxJobId;
```

Comment:
The `maxJobId` should be add `1 + maxJobId` as prefetch num. The `maxJobId` may be very large and we want to prefetch for fixed num.

Reply:
The config of `prefetch.job.num` is designed for this.